akka-sample-sharding-scala/killrweather/src/main/scala/sample/killrweather/Aggregator.scala

```diff
@@ -68,29 +68,36 @@ private[killrweather] trait Aggregator {
           context.log.debug(
             s"Recorded ${updated.size} readings from station ${entityId}, " +
               s"average ${average(updated.map(_.temperature))}, diff: processingTime - eventTime: ${received - data.eventTime}ms")
+
           replyTo ! WeatherRoutes.DataIngested(data.wsid)
 
-          aggregating(entityId, updated) // TODO store
+          aggregating(entityId, updated) // store
 
         case Query(wsid, dataType, func, replyTo) =>
-          dataType match {
+          val value = dataType match {
             case DataType.Temperature =>
               func match {
                 case Function.Average =>
-                  val value = Vector((func.tag, average(values.map(_.temperature))))
-                  replyTo ! WeatherRoutes.QueryStatus(wsid, dataType.tag, values.size, value)
+                  val start: Long = values.headOption.map(_.eventTime).getOrElse(0)
+                  val end: Long = values.lastOption.map(_.eventTime).getOrElse(0)
+                  Vector(TimeWindow(start, end, average(values.map(_.temperature))))
 
                 case Function.HighLow =>
-                  val value = Vector(
-                    (func.tag, (values.map(_.temperature).max)),
-                    (func.tag, (values.map(_.temperature).min)))
-                  replyTo ! WeatherRoutes.QueryStatus(wsid, dataType.tag, values.size, value)
-                case _ => ???
+                  val (start, min) = values.map(e => e.eventTime -> e.temperature).min
+                  val (end, max) = values.map(e => e.eventTime -> e.temperature).max
+                  Vector(TimeWindow(start, end, min), TimeWindow(start, end, max))
```

Comment:
Simplistic timestamp to accommodate min/max timestamp, Vector to allow > 1 sample aggregate.